src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -71,6 +75,50 @@ def _filter_nones(some_dict: Mapping[str, Any]) -> Dict[str, Any]:
     return {key: value for key, value in some_dict.items() if value is not None}
 
 
+@dataclass
+class SnapshotMetadata:
+    aggr_job_status_colors: defaultdict[str, dict[str, QColor]] = field(
+        default_factory=lambda: defaultdict(dict)
+    )
+    real_status_colors: dict[str, QColor] = field(default_factory=dict)
+    sorted_real_ids: list[str] = field(default_factory=list)
+    sorted_forward_model_ids: defaultdict[str, list[str]] = field(
+        default_factory=lambda: defaultdict(list)
+    )
+
+    def merge(self, other_metadata: SnapshotMetadata) -> None:
+        if other_metadata.aggr_job_status_colors:
+            self.aggr_job_status_colors.update(
+                _filter_nones(other_metadata.aggr_job_status_colors)
+            )
+        if other_metadata.real_status_colors:
+            self.real_status_colors.update(
+                _filter_nones(other_metadata.real_status_colors)
+            )
+        if other_metadata.sorted_real_ids:
+            self.sorted_real_ids = other_metadata.sorted_real_ids
+        if other_metadata.sorted_forward_model_ids:
+            self.sorted_forward_model_ids.update(
+                _filter_nones(other_metadata.sorted_forward_model_ids)
+            )
+
+    def to_dict(
+        self,
+    ) -> dict[
+        str,
+        list[str]
```

Comment:
This is a union of all the types the dataclass can contain. I am unsure of the value of typing here when we already have the types defined in the dataclass object. It might create less future maintenance if we compute the types directly here, there is something with `typing.get_type_hints(SnapshotMetadata)` according to a chatty speculative guy, it migth work.

Reply:
Dynamic type hints proved difficult. It failed with `Call expression not allowed in type expression` when trying this:
```
def to_dict(
        self,
    ) -> dict[
        str,
        get_type_hints(SnapshotMetadata),
    ]:
```

Reply:
I could however use the type hints to simplify the implementation, at the cost of readability.
How about this?
```
def to_dict(
        self,
    ) -> dict[
        str,
        list[str]
        | dict[str, QColor]
        | defaultdict[str, dict[str, QColor]]
        | defaultdict[str, list[str]],
    ]:
        hints = get_type_hints(SnapshotMetadata)
        return {field: getattr(self, field) for field in hints.keys()}
```

Reply:
nah, lets keep it as is then. It would be nicer maybe to be able to skip this to_dict and rather send the dataclass object as far as possible, but at the end it goes into Qt and it probably needs to be a pure dict.

Reply:
Hopefully, we can skip converting it to a dict altogether in the end, but I think that would have to wait until later in the dataclass refactoring project :)